### ISSUE TITLE
service.running fails to report inability to start service

### ISSUE THREAD

**Issue by aronneagu** [saltstack/salt#33540]
The minion is trying to start elasticsearch using service.running module, and the service cannot be started, but the return value of the state id is "True", while it should report false.
I know this is similar to https://github.com/saltstack/salt/issues/9959, but this doesn't seem to be a problem with the return code of "service elasticsearch status" (see below log of the minion), as it returns 3 when it fails, so saltstack should determine that the service failed to start. Please excuse me if is related with the mentioned issue.

elasticsearch version = 1.5.2
minion version = 2015.8.10
master version = 2015.8.8

```
Salt Version:
           Salt: 2015.8.8

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 2.5.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-48-generic
         system: Ubuntu 14.04 trusty
```

The state id

```
java8-pkgrepo:
  pkgrepo.managed:
    - ppa: openjdk-r/ppa
    - require_in:
      - pkg: openjdk-8-jdk

elasticsearch-pkgrepo:
  pkgrepo.managed:
    - name: deb http://packages.elastic.co/elasticsearch/1.5/debian stable main
    - key_url: https://packages.elastic.co/GPG-KEY-elasticsearch
    - require_in:
      - pkg: elasticsearch

openjdk-8-jdk:
  pkg.latest

elasticsearch:
  pkg.latest:
    - require:
      - pkg: openjdk-8-jdk
```

Output of running the state

```
          ID: start-elasticsearch
    Function: service.running
        Name: elasticsearch
      Result: True
     Comment: Service started
     Started: 11:23:18.442119
    Duration: 98.056 ms
     Changes:   
              ----------
              elasticsearch:
                  True
```

The log of the minion

```
2016-05-26 11:33:20,428 [salt.state       ][INFO    ][5682] Running state [elasticsearch] at time 11:33:20.428110
2016-05-26 11:33:20,429 [salt.state       ][INFO    ][5682] Executing state service.running for elasticsearch
2016-05-26 11:33:20,434 [salt.utils.lazy  ][DEBUG   ][5682] LazyLoaded cmd.run
2016-05-26 11:33:20,435 [salt.loaded.int.module.cmdmod][INFO    ][5682] Executing command ['runlevel', '/run/utmp'] in directory '/root'
2016-05-26 11:33:20,449 [salt.loaded.int.module.cmdmod][DEBUG   ][5682] output: N 2
2016-05-26 11:33:20,567 [salt.loaded.int.module.cmdmod][INFO    ][5682] Executing command ['service', 'elasticsearch', 'status'] in directory '/root'
2016-05-26 11:33:20,603 [salt.loaded.int.module.cmdmod][ERROR   ][5682] Command ['service', 'elasticsearch', 'status'] failed with return code: 3
2016-05-26 11:33:20,604 [salt.loaded.int.module.cmdmod][ERROR   ][5682] output:  * elasticsearch is not running
2016-05-26 11:33:20,606 [salt.loaded.int.module.cmdmod][INFO    ][5682] Executing command ['service', 'elasticsearch', 'start'] in directory '/root'
2016-05-26 11:33:20,669 [salt.loaded.int.module.cmdmod][DEBUG   ][5682] output:  * Starting Elasticsearch Server
2016-05-26 11:33:20,671 [salt.loaded.int.module.cmdmod][INFO    ][5682] Executing command ['service', 'elasticsearch', 'status'] in directory '/root'
2016-05-26 11:33:20,697 [salt.loaded.int.module.cmdmod][ERROR   ][5682] Command ['service', 'elasticsearch', 'status'] failed with return code: 3
2016-05-26 11:33:20,698 [salt.loaded.int.module.cmdmod][ERROR   ][5682] output:  * elasticsearch is not running
2016-05-26 11:33:20,700 [salt.state       ][INFO    ][5682] Started Service elasticsearch
2016-05-26 11:33:20,701 [salt.state       ][INFO    ][5682] Completed state [elasticsearch] at time 11:33:20.700547
```

**Comment by jfindlay** [comment]
@aronneagu, thanks for reporting.

**Comment by rallytime** [comment]
Hi @aronneagu - looks like this was fixed about a month ago and should be available in the upcoming 2016.3.4 release. Can you report back here once you've had a chance to test?
